compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -805,7 +805,19 @@ fn check_impl_items_against_trait<'tcx>(
                 .is_some_and(|node_item| node_item.item.defaultness(tcx).has_value());
 
             if !is_implemented && tcx.defaultness(impl_id).is_final() {
-                missing_items.push(tcx.associated_item(trait_item_id));
+                let infcx = tcx.infer_ctxt().build();
+                let ocx = ObligationCtxt::new(&infcx);
+                let param_env = tcx.param_env(trait_item_id);
```

Comment:
This is not the right param env -- it needs to be substituted with the impl's trait ref, or else it's essentially useless.